pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/ModuleListingServlet.java

```diff
@@ -115,8 +115,15 @@ protected String getQuery(SlingHttpServletRequest request) {
 
         // Module type filter
         if(!Strings.isNullOrEmpty(type)) {
-            StringBuilder moduleTypeCondition = new StringBuilder()
-                    .append("*/*/*/*/metadata/@pant:moduleType = '" + type + "'");
+            StringBuilder moduleTypeCondition;
+            if (type.equalsIgnoreCase("assembly")) {
+                moduleTypeCondition = new StringBuilder()
+                        .append("@jcr:primaryType = 'pant:assembly'");
+            } else {
+                moduleTypeCondition = new StringBuilder()
```

Comment:
Can we initialize moduleTypeCondition outside if blocks, as it will remove code duplication ?
i.e.
StringBuilder contentTypeCondition = new StringBuilder();

Reply:
done